ENGAGEHF/Education/Views/VideoPlayers/ScrollableText.swift

```diff
@@ -0,0 +1,58 @@
+//
+// This source file is part of the ENGAGE-HF project based on the Stanford Spezi Template Application project
+//
+// SPDX-FileCopyrightText: 2023 Stanford University
+//
+// SPDX-License-Identifier: MIT
+//
+
+import SwiftUI
+
+
+private struct TextSizePreference: PreferenceKey {
+    static let defaultValue: CGSize = .zero
+    
+    
+    static func reduce(value: inout CGSize, nextValue: () -> CGSize) {
+        value = nextValue()
+    }
+}
+
+
+struct ScrollableText<Background: View>: View {
+    private let text: String
+    private let background: Background
+    
+    @State private var textHeight: CGFloat = 0
+    
+    
+    var body: some View {
+        GeometryReader { geometry in
+            ScrollView {
+                Text(text)
+                    .padding()
+                    .readSize(TextSizePreference.self) {
+                        textHeight = $0.height
+                    }
+                    .frame(width: geometry.size.width, alignment: .leading)
+            }
+                .background(background)
```

Comment:
You could add a [scrollBounceBehavior(_:axes:)](https://developer.apple.com/documentation/swiftui/view/scrollbouncebehavior(_:axes:)) to ensure that it only bounces if it exceeds the size of the surrounding view. Otherwise you could "scoll" (or rather bounce) on a text that should not be scrollable.